src/proxy.ts

```diff
@@ -13,15 +13,32 @@ router.get('/ipfs/*', async (req, res) => {
   try {
     const cache = await get(`cache/${key}`);
     if (cache) return res.json(cache);
-    const json = await Promise.any(
-      gateways.map(gateway => {
-        const url = `https://${gateway}${req.originalUrl}`;
-        return fetch(url).then(res => res.json());
-      })
-    );
-    res.json(json);
-    const size = Buffer.from(JSON.stringify(json)).length;
-    if (size <= MAX) await set(`cache/${key}`, json);
+
+    let json;
+
+    try {
+      json = await Promise.any(
+        gateways.map(async gateway => {
+          const url = `https://${gateway}${req.originalUrl}`;
+          const response = await fetch(url, { timeout: 15e3 });
+
+          if (!response.ok) {
+            return Promise.reject(response.status);
+          }
+          return response.json();
+        })
+      );
+      res.json(json);
+    } catch (e: any) {
+      return res.status(400).json();
+    }
+
+    try {
+      const size = Buffer.from(JSON.stringify(json)).length;
+      if (size <= MAX) await set(`cache/${key}`, json);
+    } catch (e: any) {
+      capture(e);
```

Comment:
In this particular case, we need to throw an error again that we need to get to the outer catch
```suggestion
      capture(e);
      throw e
```

Reply:
Try/catch is there to let the AWS `set` function fail silently, such as when credentials are missing.